llmfoundry/data/finetuning/dataloader.py

```diff
@@ -192,6 +203,12 @@ def _validate_config(dataset_cfg: DictConfig):
                 'Those keys are used when building from a streaming dataset, but ' +\
                 'setting `hf_name` instructs the dataset to build from a HuggingFace dataset.'
             )
+        if '://' in dataset_cfg.hf_name.get('hf_name'):
+            if dataset_cfg.get('split') is None:
```

Comment:
Recommend moving this error check into the start of the first `if '://' in cfg.dataset.hf_name:` block.

Reply:
Oops, ignore this! Github's presentation of the code had me totally backwards on where this was getting added. This looks correct to me!